addons/assignGear/functions/fnc_assignGearVehicle.sqf

```diff
@@ -90,36 +94,276 @@ private _transportMagazines = getArray(_path >> "TransportMagazines");
+            (_loadoutInfo getOrDefault [_classname, [0, _config call _getMassLbs, getText (_config >> "displayName")]]) params ["_existingAmount", "_mass", "_name"];
+            _loadoutInfo set [_classname, [_existingAmount + parseNumber _amount, _mass, _name]];
+            nil
+        } count _transportWeapons; // count used here for speed, make sure nil is above this line
+
+        // transportBackpacks
+        {
+            (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
+            _availableBackpacks pushBackUnique _classname;
+            private _config = configFile >> "CfgBackpacks" >> _classname;
+            (_loadoutInfo getOrDefault [_classname, [0, _config call _getMassLbs, getText (_config >> "displayName")]]) params ["_existingAmount", "_mass", "_name"];
+            _loadoutInfo set [_classname, [_existingAmount + parseNumber _amount, _mass, _name]];
+            nil
+        } count _transportBackpacks; // count used here for speed, make sure nil is above this line
+
+        while {
+            _availableBackpacks isNotEqualTo [] ||
+            { _availableItems isNotEqualTo [] } ||
+            { _availableWeapons isNotEqualTo [] } ||
+            { _availableMagazines isNotEqualTo [] }
+        } do {
+            scopeName "item_select_loop";
+            private _currentWeight = 0;
+            private _box = createVehicle [_transportBoxClassname, [0, 0, 0], [], 0, "NONE"];
+            clearWeaponCargoGlobal _box;
+            clearMagazineCargoGlobal _box;
+            clearItemCargoGlobal _box;
+            clearBackpackCargoGlobal _box;
+            [_box, 0] call ace_cargo_fnc_setSize;
+            [_box, _theVehicle, true] call ace_cargo_fnc_loadItem;
 
-// transportMagazines
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    _theVehicle addMagazineCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportMagazines; // count used here for speed, make sure nil is above this line
-
-// transportItems
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    _theVehicle addItemCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportItems; // count used here for speed, make sure nil is above this line
-
-// transportWeapons
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    private _disposableName = [cba_disposable_LoadedLaunchers, _classname, "get", ""] call FUNC(getDisposableInfo);
-    if (_disposableName != "") then {
-        TRACE_2("cba_disposable_LoadedLaunchers replace",_classname,_disposableName);
-        _classname = _disposableName;
+            private _name = [];
+            if (_currentWeight < _transportBoxWeight && _availableBackpacks isNotEqualTo []) then {
+                scopeName "add_Backpack";
+                private _chosen = "";
+                private _amounts = [0];
+                for "_i" from 0 to BACKPACKS_PER_BOX do {
```

Comment:
Is there a reason to hard code the backpack limit? Or even to have one?

Reply:
design choice: because backpacks and weapons weigh the most, you will end up with boxes that are primarily bags/weapons with a few magazines and then boxes that are primarily magazines/items. if most ammo was distributed in groups of 5, where the majority of the box was a launcher, that would be unfun

Reply:
sounds good